src/index.js

```diff
@@ -636,112 +629,8 @@ export class ULabel {
-                        class="${that.config["canvas_class"]} ${that.config["imgsz_class"]} canvas_cls" 
-                        height=${that.config["image_height"] * this.config["px_per_px"]} 
-                        width=${that.config["image_width"] * this.config["px_per_px"]} 
-                        oncontextmenu="return false"></canvas>
-                    <div id="dialogs__${st}" class="dialogs_container"></div>
-                </div>
-                `);
-                $("#" + that.config["container_id"] + ` div#fad_st__${st}`).append(`
-                    <div id="front_dialogs__${st}" class="front_dialogs"></div>
-                `);
-
-                // Get canvas contexts
-                that.subtasks[st]["state"]["back_context"] = document.getElementById(
-                    that.subtasks[st]["canvas_bid"],
-                ).getContext("2d");
-                that.subtasks[st]["state"]["front_context"] = document.getElementById(
-                    that.subtasks[st]["canvas_fid"],
-                ).getContext("2d");
-            }
-
-            // Create the annotation canvases for the resume_from annotations
-            initialize_annotation_canvases(that);
-
-            // Add the ID dialogs' HTML to the document
-            build_id_dialogs(that);
-
-            // Add the HTML for the edit suggestion to the window
-            build_edit_suggestion(that);
-
-            // Add dialog to show annotation confidence
-            build_confidence_dialog(that);
-
-            // Create listers to manipulate and export this object
-            create_ulabel_listeners(that);
-
-            that.handle_toolbox_overflow();
-
-            // Set the canvas elements in the correct stacking order given current subtask
-            that.set_subtask(that.get_current_subtask_key());
-
-            that.create_overlays();
-
-            // Indicate that the object is now init!
-            that.is_init = true;
-            $(`div#${this.config["container_id"]}`).css("display", "block");
-
-            this.show_initial_crop();
-            this.update_frame();
-
-            // Draw demo annotation
-            that.redraw_demo();
-
-            // Draw resumed from annotations
-            that.redraw_all_annotations();
-
-            // Update class counter
-            that.toolbox.redraw_update_items(that);
-
-            // Call the user-provided callback
-            callback();
-        }).catch((err) => {
-            console.log(err);
-            log_message(
-                "Failed to load images: " + JSON.stringify(err),
-                LogLevel.ERROR,
-            );
-        });
-
-        // Final code to be called after the object is initialized
-        this.after_init();
-
-        console.log(`Time taken to construct and initialize: ${Date.now() - this.begining_time}`);
+        // ulabel_init(this, callback);
+        staggered_ulabel_init(this, callback, 250);
```

Comment:
any reason to keep the `ulabel_init` method? only thing i can think of is if we want to expose an option in the config to disable the loader

Reply:
The delays are really only for visualization, but we could consolidate to a single method that just sets the delays to 0 in the default case. 
We could also consolidate to one method without delays. 

Alternative to all of the above, we could have a minimum fixed loading time to always show off the loader, picking and choosing which components visually render in behind the overlay. There is some validity to artificial delays for UX (see [this](https://90percentofeverything.com/2010/12/16/adding-delays-to-increase-perceived-value-does-it-work/)), but I definitely would want that to be configurable - there is at least one ULabel user that would prefer instantaneous loading (me).

Reply:
yeah i dont think we need to add an option for a fixed loading time. one thought i had is that perhaps we could support certain config args to be Promises and then we can await them in the init so that the loader is more versatile. IE if a user is hitting an API to retrieve an image url, they could pass a promise to ulabel and then the loader could start before the image promise resolves. thoughts?

Reply:
Do you think for now I should keep the async one, bonk the old one and all the delays?

Reply:
if the delays were just for demoing the loader then for now we can remove them all or use the old init. then we should look into allowing users to choose when to start the loader, like maybe we provide a static method that allows the user to start the loader before even calling the ulabel constructor, and that same method that runs in the html_bulider would check if a loader is already present before adding one

Reply:
although if the ulabel html and styles arent present yet that might not work actually

Reply:
Yeah what I noticed while testing this is that the ULabel container is intentionally not visible for a good portion of the loading process, so in `staggered_ulabel_init` I moved the step that makes it visible to be way earlier to show what's actually going on. 

We could just tweak the container hierarchy slightly so that the loader is attached to the main div immediately, and everything else is attached to another div that waits a bit before displaying. That would be set up in a way to allow it to be launched earlier if desired. 

Another consideration is how the `.on` callback utility works with async functions. Consider:
```javascript
async function fn() {
...
}
ulabel.on(ulabel.fn, () => { ... })
// Does this await just fn, or also it's attached callbacks?
await ulabel.fn();
```
Easy enough to test, and if it doesn't it should be possible to add that functionality through a different method. 
If we got that working, we could just structure initialization steps together and let the user attaching loading logic wherever they want, including when to display the loader.

Reply:
yeah after thinking about it a bit, i think in my head i want two uses for the loader. First is what is implemented here, which is a loader that displays as ulabel performs its init. Second is a set of static methods so that a user can choose when to show the same style of loader while they are doing whatever prep is needed before calling ulabel's constructor and init, like loading the image and annotations etc. We could split the second way off into its own issue and release what we have here for now.

Reply:
For the most part this could use the same logic. Taking inspiration from [Textual](https://textual.textualize.io/), where every widget has a reactive `.loading` attribute which will add a loading icon and an overlay, we could do something similar. We would just have to refactor the logic to be entirely separate from the ULabel constructor, but also called as part of the normal `init` logic.

Reply:
Opened #206 to track this.